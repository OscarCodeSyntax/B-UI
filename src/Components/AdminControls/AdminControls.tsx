import { Button, FormControl, FormLabel, TextField } from "@mui/material";
import React, { useState } from "react";
import UserLoginQueries from "../../Resources/Queries/UserLoginQueries";

function AdminControls() {
  const [username, setUsername] = useState("");

  return (
    <div className="padding-10">
      <FormControl>
        <FormLabel>Username to delete</FormLabel>
        <TextField
          type="username"
          onChange={(e) => setUsername(e.target.value)}
          name="username"
          value={username}
        />
        <Button
          className="width-250px"
          variant="outlined"
          onClick={(e: any) => {
            UserLoginQueries.deleteUser(username);
          }}
          title="Find By Id"
          value="Find By Id"
        >
          Delete User
        </Button>
      </FormControl>
    </div>
  );
}

export default AdminControls;
