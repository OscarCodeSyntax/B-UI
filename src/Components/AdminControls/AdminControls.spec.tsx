import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import AdminControls from "./AdminControls";

describe("Admin Controls ", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly", () => {
    const AdminControlsWrapper = renderer.create(<AdminControls />).toJSON();
    expect(AdminControlsWrapper).toMatchSnapshot();
  });

  it("renders  1 input element", () => {
    const testRenderer = renderer.create(<AdminControls />);
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("input").length).toEqual(1);
  });

  it("renders  1 button element", () => {
    const testRenderer = renderer.create(<AdminControls />);
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("button").length).toEqual(1);
  });

  it("has all the correct text", () => {
    render(<AdminControls />);
    expect(screen.getByText("Username to delete")).toBeTruthy();
    expect(screen.getByText("Delete User")).toBeTruthy();
  });
});
