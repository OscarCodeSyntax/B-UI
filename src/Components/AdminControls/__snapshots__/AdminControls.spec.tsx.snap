// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Controls  renders correctly 1`] = `
<div
  className="padding-10"
>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
    >
      Username to delete
    </label>
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="username"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium width-250px css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Find By Id"
      type="button"
      value="Find By Id"
    >
      Delete User
    </button>
  </div>
</div>
`;
